app/helpers/tags_helper.rb

```diff
@@ -1,40 +1,47 @@
 # frozen_string_literal: true
 module TagsHelper
-  def tag_usage(data_id, data_type, user_id)
-    result = []
-    moments = User.find_by(id: user_id).moments.order('created_at DESC')
-    if data_type == 'category'
-      result = get_moments(data_id, data_type, user_id, moments)
-    elsif data_type.in?(%w[mood strategy])
-      result = get_moods(data_id, data_type, moments)
-    end
-    result
+  include MomentsHelper
+
+  def tagged_moments_data_json
+    { data: moments_or_strategy_props(@moments),
+      lastPage: @moments.last_page? }
+  end
+
+  def tagged_strategies_data_json
+    { data: moments_or_strategy_props(@strategies),
+      lastPage: @strategies.last_page? }
+  end
+
+  def setup_stories
+    return unless viewable_tag?(@category) || viewable_tag?(@mood) ||
+                  viewable_tag?(@strategy)
+
+    @moments = get_tagged_data(
+      @category || @mood || @strategy,
+      User.find_by(id: current_user.id).moments.recent
+    )
+    @strategies = get_tagged_data(
+      @category, User.find_by(id: current_user.id).strategies.recent
+    )
   end
 
   private
 
-  def get_moods(data_id, data_type, moments)
-    result = []
-    moments.find_each do |m|
-      result << m if data_included?(data_type, data_id, m)
-    end
-    result
+  def viewable_tag?(data)
+    return unless data
+
+    data.user_id == current_user.id ||
+      (data.try(:viewers)&.include?(current_user.id) &&
+        data.published?)
   end
 
-  def get_moments(data_id, data_type, user_id, moments)
+  def get_tagged_data(tag, data)
+    return unless tag
```

Comment:
Thinking out aloud here. wouldn't we add `return unless` for both params
```suggestion
    return unless tag && data
```